Tools/OrcEnginePhase0/oracle/manifest.py

```
@@ -0,0 +1,152 @@
+
+
+def _git_commit() -> str:
+    """Real commit SHA at manifest-generation time, not a 'see git log' placeholder --
+    this is what actually binds a manifest's tensor checksums to the exact oracle/model.py
+    revision that generated them (CodeRabbit finding, PR #102)."""
+    try:
+        out = subprocess.run(["git", "rev-parse", "HEAD"], capture_output=True, text=True,
+                              cwd=os.path.dirname(__file__), timeout=10)
+        sha = out.stdout.strip()
+        if out.returncode == 0 and sha:
+            dirty = subprocess.run(["git", "status", "--porcelain"], capture_output=True, text=True,
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)